hangman.rb

```diff
@@ -0,0 +1,21 @@
+# define a method name hangman with 2 parameters
+def hangman(word, guess)
+   # empty array
+   output = []
+   # use split to split word into individual characters
+   letters = word.split('')
+   # .map transforms empty array from values returned from the block
+   letters.map do |letter|
```

Comment:
```suggestion
   letters.each do |letter|
```

The `map` method creates a new collection with transformed values and returns it, but you already created your own collection with the `output = []`. You don't need the map at all; for more details, see <https://www.rubyguides.com/2018/10/ruby-map-method/>

Bonus exercise: if you truly wanted to use `map`, you can do something like `output = word.split('').map do |letter|`... and the code will actually come out much more compact (and nice IMO). I'll leave the decision of whether to do that work up to you.